How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,38 @@
+
+Data comes from multiple sources and is ingested at different intervals; annually from trusted sources (census data), a single time (Great Schools), or created by CCAO employees (ccao-condominium-pin_condo_char). Complete this task in R or Python, keeping data "raw", i.e. close to it's original structure. This file can be uploaded to the correct [S3 bucket](https://us-east-1.console.aws.amazon.com/s3/home?region=us-east-1#) in either parquet or excel format.
+
+### Step 2 - Transformation into [etl/scripts-ccao-data-warehouse-us-east-1](https://github.com/ccao-data/data-architecture/tree/master/etl/scripts-ccao-data-warehouse-us-east-1)
+
+Raw data often includes unwanted variables, column names that don't align with CCAO coding practices, or data outside of Cook County / IL. Clean the data in Python so that it can be run through DBT / Github actions. Upon completion of data cleaning, make sure that data is uploaded in Parquet format to the correct S3 Bucket.
+
+## Structure B - Use SQL and Existing DBT Macros
+
+Over time, CCAO has developed a series of [macro functions](https://github.com/ccao-data/data-architecture/tree/master/dbt/macros) which implement standardized transformations. These use existing resources (i.e. vw_pin_parcel) in AWS to create new features, often utilizing spatial joins.
+
+### Step 1 - Create a SQL Query in [data_architecture/dbt/models](https://github.com/ccao-data/data-architecture/tree/master/dbt/models)
+
+Take existing data from one of the CCAO's AWS buckets, create a SQL query that utilizes one of the macro functions, and write that into a configured parquet table. This will then be uploaded to an S3 bucket.
+
+## Collective Steps
+
+### Step 1 - Run Crawler for Ingest into AWS Athena.
+Once final data has been uploaded to the AWS S3, it needs to be transfered to [AWS Athena](https://us-east-1.console.aws.amazon.com/athena/home?region=us-east-1#/query-editor) via the [crawler](https://us-east-1.console.aws.amazon.com/glue/home?region=us-east-1#/v2/data-catalog/crawlers). Find the correct bucket, and then click 'Run Crawler' in the upper right. 
+
+### Step 2 - Include in the Model Pipeline
+
+To include the feature in the model pipeline, the sql queries vw_pin_condo_input, vw_res_card_input, and vw_shared_input located in [data_architecture/dbt/models/model/](https://github.com/ccao-data/data-architecture/tree/master/dbt/models/model) need to be modified.
+
+### Step 3 - Update Documentation
+
+CCAO data is largely mantained through DBT. To ensure data integrity, documentation needs to be updated alongside the new feature. These can be found in data_architecture/dbt/models/model/schema.yml, dbt/models/shared_columns.md, dbt/models/.../docs.md, and dbt/models/.../schema.yml.
+
+## Final Step - An Optional Horrah
+
+Test the impact of the feature by running the model via the following [guide](https://github.com/ccao-data/model-res-avm/blob/master/README.md). 
```

Comment:
**[Suggestion, blocking]** I think the "Structure A / Structure B / Collective Steps" organizational outline could be a bit clearer, and I think we're missing a couple of steps toward the end that tie the input views to the model itself. Here's how I would outline the process:

1. Extract and load raw data
    * Do this by adding a raw data extraction script to `etl/scripts-ccao-data-raw-us-east-1/` using Python or R
    * Choose the output location based on the transformation strategy in step 2: The script should push data to `ccao-data-warehouse-us-east-1` if transforming the data using a dbt model, or `ccao-data-raw-us-east-1` if transforming the data using a Python/R script
2. Transform the raw data
    * Option 1: [Add a model to the dbt DAG](https://github.com/ccao-data/data-architecture/blob/master/dbt/README.md#-how-to-add-a-new-model) to transform the data
        * This requires that the raw data you pushed in step 1 live in the `ccao-data-warehouse-us-east-1` bucket, be crawled by Glue, and be configured as a [source](https://docs.getdbt.com/docs/build/sources) in a dbt `schema.yml` file
        * This is a good option if your transformation is simple enough to be written as a SQL query, or as a Python script using Athena PySpark's [built-in third-party packages](https://docs.aws.amazon.com/athena/latest/ug/notebooks-spark-preinstalled-python-libraries.html)
    * Option 2: Add a Python/R script to `etl/scripts-ccao-data-warehouse-us-east-1/`
        * This requires that the raw data you pushed in step 1 live in the `ccao-data-raw-us-east-1` bucket
        * This is a good option if your transformation is complex enough to require a Python or R script using third-party packages beyond what is built-in to Athena PySpark
        * If you choose this option, you'll need to configure your output table as a [source](https://docs.getdbt.com/docs/build/sources) in a dbt `schema.yml` file and set up a Glue crawler for it
3. Incorporate the transformed data into the model input views
    * The views in question include `vw_pin_condo_input`, `vw_res_card_input`, and `vw_shared_input`; choose the correct view depending on whether the feature should be available to the condo model, the res model, or both, and update the view definition in the appropriate dbt model to join to the transformation table you created in step 2
4. Update each model to use the new versions of the model input views
    * For each model:
        1. Rerun the model ingest steps to pull from the new versions of the input views
        2. Push the updated model input data to DVC
        3. Update the `predictor` attribute in `params.yml` to add the new columns as predictors